providers/apis/coingecko/utils.go

```diff
@@ -21,6 +21,13 @@ const (
 	// an API key but may be rate limited.
 	URL = "https://api.coingecko.com/api/v3"
 
+	// APIURL is the base URL for the CoinGecko API. This URL requires an API
+	// key and is not rate limited.
+	APIURL = "https://pro-api.coingecko.com/api/v3"
+
+	// APIKeyHeader is the header used to pass the API key to the CoinGecko API.
+	APIKeyHeader = "x-cg-pro-api-key" //nolint
```

Comment:
Can't you just configure this in the Endpoint and delete all the WithHTTPHeaders stuff?

Reply:
im just keeping this here for reference mostly, also useful to have in case you want to test locally.